libbeat/common/kubernetes/watcher.go

```diff
@@ -58,6 +58,10 @@ type WatchOptions struct {
 	Node string
 	// Namespace is used for filtering watched resource to given namespace, use "" for all namespaces
 	Namespace string
+	// LabelsDedot is used for replacing dots in labels with `_`
+	LabelsDedot bool
```

Comment:
watcher is not really using these settings, perhaps you can move them to the MetricSet struct

Reply:
@exekias Good point! I moved these two into a seperate config struct which is included in MetricSet. Is this way better? Thanks! https://github.com/elastic/beats/blob/a95a5748181e9c494c1d7a06624b086afae1d7c3/metricbeat/module/kubernetes/event/event.go#L51